libs/azure-storage/tests/unit_tests/test_document_loaders.py

```diff
@@ -0,0 +1,160 @@
+import azure.identity
+import pytest
+from azure.storage.blob import ContainerClient
+from pytest_mock import MockerFixture
+
+from langchain_azure_storage.document_loaders import AzureBlobStorageLoader
+
+
+@pytest.fixture
+def blob_names() -> list[str]:
+    return ["text_file.txt", "json_file.json", "csv_file.csv"]
+
+
+@pytest.fixture
+def blob_contents() -> list[str]:
+    return [
+        "test content",
+        "{'test': 'test content'}",
+        "col1,col2\nval1,val2",
+    ]
+
+
+def _blob_storage_loader_args() -> dict:
+    return {
+        "account_url": "https://testaccount.blob.core.windows.net",
+        "container_name": "test-container",
+    }
+
+
+@pytest.fixture
+def mock_lazy_load_documents_from_blob(
+    mocker: MockerFixture, blob_names: list[str], blob_contents: list[str]
+) -> None:
+    from langchain_core.documents.base import Document
+
+    docs = []
+    for name, content in zip(blob_names, blob_contents):
+        docs.append(
+            Document(
+                page_content=content,
+                metadata={
+                    "source": f"https://testaccount.blob.core.windows.net/test-container/{name}"
+                },
+            )
+        )
+    mocker.patch(
+        "langchain_azure_storage.document_loaders.AzureBlobStorageLoader._lazy_load_documents_from_blob",
+        return_value=iter(docs),
+    )
+
+
+def test_lazy_load(
+    mocker: MockerFixture,
+    blob_names: list[str],
+    blob_contents: list[str],
+    mock_lazy_load_documents_from_blob: None,
+) -> None:
+    mock_container_client = mocker.patch.object(ContainerClient, "from_container_url")
+    mock_client = mocker.MagicMock()
+    mock_client.list_blob_names.return_value = blob_names
+    mock_container_client.return_value = mock_client
+
+    args = {**_blob_storage_loader_args(), "blob_names": blob_names}
+    loader = AzureBlobStorageLoader(**args)
```

Comment:
It might be worth just making a `create_azure_blob_storage_loader()` pytest fixture. Sort of similar to what we do [here](https://github.com/Azure/azure-storage-for-pytorch/blob/40cd9b66eaa0a9214553dc7f4e69ebf813063e5e/tests/unit/test_datasets.py#L19) in the PyTorch Connector. Mainly it is advantageous because we can then expose things like `account_url` and `container_name` as fixtures and can then be directly asserted against instead of having to reduplicate these values.